tests/unit/storage/createRestStorage.ts

```diff
@@ -0,0 +1,812 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import Task from 'dojo-core/async/Task';
+import { Handle } from 'dojo-interfaces/core';
+import Map from 'dojo-shim/Map';
+import {providerRegistry, RequestOptions, Response, ResponsePromise } from 'dojo-core/request';
+import { StoreOperation } from '../../../src/store/createStore';
+import createRestStorage from '../../../src/storage/createRestStorage';
+import { Storage } from '../../../src/storage/createBaseStorage';
+import { createData, patches, patchedItems } from '../support/createData';
+import { createSort, Sort } from '../../../src/query/Sort';
+import Filter, { createFilter } from '../../../src/query/Filter';
+import StoreRange, { createRange } from '../../../src/query/StoreRange';
+import createCompoundQuery, { Query }  from '../../../src/query/createQuery';
+
+const handles: Handle[] = [];
+const responses: Response<any>[] = [];
+
+interface MockResponse {
+	response?: string;
+	headers?: { [index: string]: string };
+}
+const emptyResponse: MockResponse = { response: null, headers: {} };
+
+function registerMockProvider(pattern: string | RegExp, mockResponses: Map<string, MockResponse> = new Map<string, {}>() ) {
+	return providerRegistry.register(pattern, function(url: string, options: RequestOptions): ResponsePromise<any> {
+		const mockResponse = mockResponses.get(options.method) || emptyResponse;
+		if (mockResponse instanceof Error) {
+			return Task.reject(mockResponse);
+		}
+		else {
+			const response: Response<any> = {
+				data: mockResponse.response,
+				nativeResponse: null,
+				requestOptions: options,
+				statusCode: 200,
+				statusText: 'mock status',
+				url,
+				getHeader(name: string) {
+					return mockResponse.headers[name];
+				}
+			};
+			responses.push(response);
+			return <ResponsePromise<any>> Task.resolve(response);
+
+		}
+	});
+}
+
+let storage: Storage<{}, {}>;
+
+function clear() {
+	handles.forEach(function(handle) {
+		if (handle) {
+			handle.destroy();
+		}
+	});
+	handles.length = 0;
+	responses.length = 0;
+}
+
+registerSuite({
+	name: 'createRestStorage',
+
+	afterEach: clear,
+
+	beforeEach() {
+		handles.push(registerMockProvider(/\/mockItems.*$/));
+		storage = createRestStorage({
+			target: '/mockItems'
+		});
+	},
+
+	'GET': {
+		'ids can\'t be empty'() {
```

Comment:
Can you be a bit more descriptive with your names please. ie.
`Should throw error when get called without ids`